app/models/address.rb

```diff
@@ -41,14 +41,38 @@ def self.geocode_all_needed(sleep_between: 0.5, num_per_batch: 50)
       addr.save
       sleep(sleep_between.to_f)
     end
-    
+
   end
 
+  def address_array
+    # This should only be called for address associated with a company
 
-  def entire_address
-    [street_address, city, post_code, sverige_if_nil].compact.join(', ')
+    # Returns an array of address field values (strings) that starts with
+    # with address_visibility level set for the associated company.
+
+    visibility_level = addressable.address_visibility
+
+    address_pattern = %w(street_address post_code city kommun)
+
+    start_index = address_pattern.find_index do |field|
+      field == visibility_level
+    end
+
+    return [] unless start_index
+
+    if kommun
+      ary = [street_address, post_code, city, kommun.name,
+             sverige_if_nil][start_index..4]
+    else
+      ary = [street_address, post_code, city,
+             sverige_if_nil][start_index..3]
+    end
```

Comment:
Should `4` and `3` above be constants? Or are they related to the length of a`ary` -- in which case they could be variable names, just so that the intent is clearer